macosx/FilterBarView.m

```diff
@@ -52,14 +52,22 @@ - (BOOL) isOpaque
 - (void) drawRect: (NSRect) rect
 {
     if ([NSApp isOnYosemiteOrBetter]) {
-        [[NSColor windowBackgroundColor] setFill];
-        NSRectFill(rect);
-
-        const NSRect lineBorderRect = NSMakeRect(NSMinX(rect), 0.0, NSWidth(rect), 1.0);
-        if (NSIntersectsRect(lineBorderRect, rect))
-        {
-            [[NSColor lightGrayColor] setFill];
-            NSRectFill(lineBorderRect);
+        NSColor * lineColor = [NSColor lightGrayColor];
+        NSColor * color = [NSColor windowBackgroundColor];
+        switch ([NSApp willUseSemanticColors]) {
```

Comment:
Why is this a switch statement? Instead of an if to change the colors. Any why put any of the "case 0" in the switch since everything will hit this.

Reply:
"Case 0" is meant to hit regardlessly because it draws the bottom border while the switch is to determine if it's necessary to use semantic colors to avoid potential crashes (due to MacOS X versions lacking these colors) but allow them on newer versions of macOS. I see how I could use a switch instead to change the colors and get rid of one duplicate variable to save some lines. Thanks you input.

Reply:
Ya, just in general you should never switch on BOOLs (just enums really in obj-c), and you should never (well, almost never) use the fall through intentionally. Put setting the new colors in the if-statement and the code you always want to hit outside the if-statement.